src/main/java/net/minecraftforge/client/event/CommandRegistryEvent.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2020.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.event;
+
+import com.mojang.brigadier.CommandDispatcher;
+import net.minecraft.command.ISuggestionProvider;
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ * Fired when client {@link CommandDispatcher} has all of its commands synced from the server
+ */
+public class CommandRegistryEvent extends Event
```

Comment:
As we have a `RegisterCommandsEvent`, it might be good to call this something similar like `RegisterClientCommandsEvent` and have similar context. I would, however, not extend `RegisterCommandsEvent` so that people that listen for `RegisterCommandsEvent` don't accidentally register for the client as well

Reply:
The event you refer to has a very different purpose (and was proposed later than this).
I would argue it to be less confusing when the name are not too similar.

Reply:
How this this event different in purpose from the RegisterCommandsEvent? Both are intended to be used to register new commands, one for client-side commands, one for server-side commands. There are fired at different times and have a completly different implementation, but to the modder they serve the same purpose for different sides or am I wrong?

Reply:
The RegisterCommandsEvent is there to fix registration originating from vanilla datapacks, and triggers on both client and server side. The registered command won't execute on the client and may or may not execute on the server, depending on various conditions.

While this PR is intended to be a long-term feature supported by Forge, and a new instalment of what was supported in older versions a.k.a _client-only_ 'registered **and executed**' commands, the execution is guaranteed not conditional.
And as you said, they trigger at different times so if a modder used those for a generic listener, he would not get the same behaviour with both event.